bundles/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/internal/handler/PHCBridgeHandler.java

```diff
@@ -104,215 +110,405 @@ public void initialize() {
-                    // Can´t process the acknowledgement of JRM yet.
-                    logger.info("No acknowlgdge from the module {} received.", qo.getModuleAddress());
+                try {
+                    Thread.sleep(SEND_RETRY_TIME_MILLIS);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
                 }
+            } while (!isChannelOutputState(qo.getModuleType(), qo.getModuleAddress(), qo.getChannel(), qo.getCommand())
+                    && sendCount < SEND_RETRY_COUNT);
+
+            if (qo.getModuleType().equals(PHCBindingConstants.CHANNELS_JRM)) {
+                // there aren't state per channel for JRM modules
+                amOutputState[qo.getModuleAddress() & 0x1F] = -1;
+            } else if (qo.getModuleType().equals(PHCBindingConstants.CHANNELS_DIM)) {
+                // state ist the same for every dim level except zero/off -> inizialize state
+                // with 0x0F after sending an command.
+                dmOutputState[qo.getModuleAddress() & 0x1F] |= (0x0F << (qo.getChannel() * 4));
             }
 
+            if (sendCount >= SEND_RETRY_COUNT) {
+                // change the toggle: if no acknowledge received it may be wrong.
+                byte module = qo.getModuleAddress();
+                if (PHCBindingConstants.CHANNELS_AM.equals(qo.getModuleType())
+                        || PHCBindingConstants.CHANNELS_JRM.equals(qo.getModuleType())) {
+                    module |= 0x40;
+                } else if (PHCBindingConstants.CHANNELS_DIM.equals(qo.getModuleType())) {
+                    module |= 0xA0;
+                }
+                toggleMap.put(module, !getToggle(module));
+
+                if (logger.isDebugEnabled()) {
+                    logger.debug("No acknowledge from the module {} received.", qo.getModuleAddress());
+                }
+            }
         }
     }
 
-    private void setModuleOutputState(String moduleType, byte moduleAddress, byte state) {
+    private void setModuleOutputState(String moduleType, byte moduleAddress, byte state, byte channel) {
         if (moduleType == PHCBindingConstants.CHANNELS_EM_LED) {
             emLedOutputState[moduleAddress] = state;
         } else if (moduleType == PHCBindingConstants.CHANNELS_AM) {
             amOutputState[moduleAddress & 0x1F] = state;
-        } else if (moduleType == PHCBindingConstants.CHANNELS_JRM) {
-            // not implemented yet
+        } else if (moduleType == PHCBindingConstants.CHANNELS_DIM) {
+            dmOutputState[moduleAddress & 0x1F] = (byte) (state << channel * 4);
         }
     }
 
-    private State isChannelOutputStateSet(String moduleType, byte moduleAddress, byte channel) {
-        State set = null;
+    private boolean isChannelOutputState(String moduleType, byte moduleAddress, byte channel, Command cmd) {
+        int state = cmd.equals(OnOffType.OFF) ? 0 : 1;
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)